docs/developer/app-store/apps/adyen.mdx

```diff
@@ -0,0 +1,355 @@
+            "environment": "TEST"
+          },
+          "errors": []
+        }
+      ],
+      "errors": []
+    }
+  }
+}
+```
+
+:::tip
+
+For instructions on how to add, remove or constraint payment methods from Adyen, please consult the [Adyen payment methods documentation](https://docs.adyen.com/payment-methods).
+
+:::
+
+### Paying with Adyen
+
+After a user has interacted with the Adyen Drop-in and entered payment details, [Drop-in event data along with other information](https://docs.adyen.com/online-payments/web-drop-in/additional-use-cases/advanced-flow#step-3-make-a-payment) should be passed to the [`transactionInitialize`](../../../api-reference/payments/mutations/transaction-initialize) mutation as the `paymentGateway.data` field. The mutation returns the `TransactionInitialize` object with a `data` field containing the following fields:
+
+```ts
+{
+  paymentResponse: PaymentResponse;
+  errors?: SyncWebhookAppErrors;
+}
+```
+
+Where `PaymentResponse` is the result of calling Adyen's `/payments` endpoint and is described in the [Adyen documentation](https://docs.adyen.com/api-explorer/Checkout/latest/post/payments#responses-200). `SyncWebhookAppErrors` is [described below](#handling-errors).
+
+If the `errors` field doesn't exist or is an empty array, pass the `paymentResponse` to Adyen Drop-in. The Drop-in will handle the response and display the result to the user or require additional actions to proceed.
+
+```graphql
+mutation AdyenTransactionInitialize($data: JSON!) {
+  transactionInitialize(
+    id: "Q2hlY2tvdXQ6YWY3MDJkMGQtMzM0NC00NjMxLTlkNmEtMDk4Yzk1ODhlNmMy"
+    action: AUTHORIZATION
+    amount: 54.24
+    paymentGateway: { id: "app.saleor.adyen", data: $data }
+  ) {
+    transactionEvent {
+      pspReference
+      amount {
+        amount
+        currency
+      }
+      type
+    }
+    data
+    errors {
+      field
+      message
+      code
+    }
+  }
+}
+```
+
+Where `$data` is the object provided by Adyen Drop-in in the `onSubmit` (Web, React Native), `didSubmit` (iOS) or `makePaymentsCall` (Android) callback. Response:
+
+```json
+{
+  "data": {
+    "transactionInitialize": {
+      "transactionEvent": {
+        "pspReference": "XXXX9XXXXXXXXX99",
+        "amount": {
+          "amount": 54.24,
+          "currency": "EUR"
+        },
+        "type": "AUTHORIZATION_SUCCESS"
+      },
+      "data": {
+        "paymentResponse": {
+          "additionalData": {
+            "paymentMethod": "visa"
+          },
+          "amount": {
+            "currency": "EUR",
+            "value": 5424
+          },
+          "merchantReference": "SOME_MERCHANT_ID_",
+          "paymentMethod": {
+            "brand": "visa",
+            "type": "scheme"
+          },
+          "pspReference": "XXXX9XXXXXXXXX99",
+          "resultCode": "Authorised"
+        }
+      },
+      "errors": []
+    }
+  }
+}
+```
+
+### Performing additional actions (_optional_)
```

Comment:
Is this really optional? What users should do to make a payment when 3DS is required?